src/steamcompmgr.cpp

```diff
@@ -4611,14 +4609,28 @@ void x11_set_selection(xwayland_ctx_t *ctx, std::string contents, int selectionT
 	XFlush(ctx->dpy);
 }
 
-static void
-handle_selection_request(xwayland_ctx_t *ctx, XSelectionRequestEvent *ev)
+void gamescope_set_selection(std::string contents, int selection)
 {
-	if (!BIsNested())
+	if (selection == CLIPBOARD)
 	{
-		return;
+		clipboard = contents;
+	}
+	else if (selection == PRIMARYSELECTION)
+	{
+		primarySelection = contents;
 	}
 
+	for (int i = 0; i < g_nXWaylandCount; i++)
+	{
+		gamescope_xwayland_server_t *server = wlserver_get_xwayland_server(i);
```

Comment:
Should call this like the other things that loop with `gamescope_xwayland_server_t` and also avoid updating the atom for the server that's calling it as we don't want to end up in a race with buffered messages on the same server.

Reply:
Can you elaborate on what you mean by "like the other things that loop with `gamescope_xwayland_server_t`?"

Reply:
```
		gamescope_xwayland_server_t *server = NULL;
		for (size_t i = 0; (server = wlserver_get_xwayland_server(i)); i++)
```

Reply:
As it turns out, we shouldn't actually need to update the atom at all with this call in any of the instances, since that gets done when we receive a SelectionRequest.  All we really want to do here is become the selection owner.  Again, I'm not all that familiar with Xlib, so please let me know if I'm missing something